javapoet/src/test/java/com/palantir/javapoet/FieldSpecTest.java

```diff
@@ -19,9 +19,37 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import javax.lang.model.element.Modifier;
+import org.junit.After;
 import org.junit.Test;
 
 public class FieldSpecTest {
+    /**
+     * Field spec created by a test method; used by {@link #checkToBuilderRoundtrip()}.
+     *
+     * <p>{@code null} if a test method does not create a (valid) field spec, or if no round-trip check
+     * should be performed on it.
+     */
+    private FieldSpec fieldSpec = null;
+
+    /**
+     * Performs round-trip check that {@code fieldSpec.toBuilder().build()} is identical to the
+     * original {@code fieldSpec}.
+     */
+    @After
+    public void checkToBuilderRoundtrip() {
```

Comment:
Thoughts on making this a static utility function instead? It's not used in every test, and I think it'll be a bit easier to understand test failures if we use it as a test utility function from within the test body.

Something like this, eliminating the need for a new field:
```java
private static void checkToBuilderRoundtrip(FieldSpec fieldSpec) {
```

Then the test method may be updated thusly:
```diff
-        // Perform round-trip check in @After
-        fieldSpec = a;
+        checkToBuilderRoundtrip(a);
```

Reply:
My intention was to have this implicitly (through the field) so that if a new builder method is added (and additional data is added to the builder) and a user writes a new test for it, they implicitly get this check without having to manually call `checkToBuilderRoundtrip`.

But yes, on the other hand that approach with this field is not that clean.
I will probably completely remove these fields again and make the `...Spec` values local variables again, and then use a `static` method.

Reply:
I have refactored it now, but in some cases omitted the `checkToBuilderRoundtrip` call where it was most likely already covered by other test methods.